lib/internal/http2/core.js

```diff
@@ -724,7 +729,8 @@ class Http2Session extends EventEmitter {
     this[kSocket] = socket;
 
     // Do not use nagle's algorithm
-    socket.setNoDelay();
+    if (typeof socket.setNoDelay === 'function')
+      socket.setNoDelay();
 
     // Disable TLS renegotiation on the socket
```

Comment:
A bit further down in this code there is a check for `socket.connecting` that explicitly defers actions until the socket is connected. With this change, the assumption would be that the non-socket Stream passed in is immediately available for use, which is a perfectly fine assumption to make, IMHO, but I want to make sure that is *actually* a safe assumption here.

Reply:
I’m not sure what the alternative would be. We could document `socket.connecting`/`socket.on('connect')` as part of the implicit API here?

Reply:
That would certainly be an option, but given that we're wrapping the custom socket, the likelihood of it even being an issue is pretty low. Let's go with this for now :-)